rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    @Override
+    public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            LockWhitelistEntry entry = new UnlimitedWhiteListEntry(address);
+
+            return addLockWhitelistAddress(tx, entry);
+        } catch (AddressFormatException e) {
+            logger.warn("[addUnlimitedLockWhitelistAddress] {}", INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (!isAuthorizedLockWhitelistChange(tx)) {
+            return UNAUTHORIZED_CALLER.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            if (!whitelist.remove(address)) {
+                return ADDRESS_NOT_EXIST.getCode();
+            }
+            return SUCCESS.getCode();
+        } catch (AddressFormatException e) {
+            logger.error("[removeLockWhitelistAddress] {}", INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int setLockWhitelistDisableBlockDelay(Transaction tx, BigInteger disableBlockDelayBI, int btcBlockchainBestChainHeight) {
+
+        if (!isAuthorizedLockWhitelistChange(tx)) {
+            return UNAUTHORIZED_CALLER.getCode();
+        }
+
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist();
+        if (lockWhitelist.isDisableBlockSet()) {
+            return DELAY_ALREADY_SET.getCode();
+        }
+
+        int disableBlockDelay = disableBlockDelayBI.intValueExact();
+        if (disableBlockDelay + btcBlockchainBestChainHeight <= btcBlockchainBestChainHeight) {
+            return DISABLE_BLOCK_DELAY_INVALID.getCode();
+        }
+
+        lockWhitelist.setDisableBlockHeight(btcBlockchainBestChainHeight + disableBlockDelay);
+
+        return SUCCESS.getCode();
+    }
+
+    @Override
+    public boolean verifyLockSenderIsWhitelisted(Address senderBtcAddress, Coin totalAmount, int height) {
+        final String ADDRESS_NOT_WHITELISTED_MESSAGE = "Rejected lock. Address is not whitelisted. Address: ";
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist();
+        if (!lockWhitelist.isWhitelistedFor(senderBtcAddress, totalAmount, height)) {
+            logger.info("[verifyLockSenderIsWhitelisted] {} {}", ADDRESS_NOT_WHITELISTED_MESSAGE, senderBtcAddress);
+            return false;
+        }
+        // Consume this whitelisted address
+        lockWhitelist.consume(senderBtcAddress);
+
+        return true;
+    }
+
+    private Integer addLockWhitelistAddress(Transaction tx, LockWhitelistEntry entry) {
```

Comment:
please move this private method close to where its being used, right after `addUnlimitedLockWhitelistAddress`

Reply:
Done!